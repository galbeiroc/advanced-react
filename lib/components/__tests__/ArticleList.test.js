import React from 'react';
import renderer from 'react-test-renderer';

import ArticleList from '../article/ArticleList';

const testProps = {
  articles: {
    a: { id: 'a' },
    b: { id: 'b' }
  },
  store: {
    lookupAuthor: jest.fn(() => ({}))
  }
}

describe('ArticleList', () => {
  it('render correctly', () => {
    const tree = renderer.create(
      <ArticleList {...testProps} />
    ).toJSON();

    expect(tree).toMatchSnapshot();
    expect(tree.children.length).toBe(2);
  });
});
