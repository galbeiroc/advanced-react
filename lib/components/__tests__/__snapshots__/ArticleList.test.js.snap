// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList render correctly 1`] = `
<div>
  <ArticleWrapper
    article={
      Object {
        "id": "a",
      }
    }
  />
  <ArticleWrapper
    article={
      Object {
        "id": "b",
      }
    }
  />
</div>
`;
